File: fsharp/darts/Darts.fs
module Darts

[<Literal>]
let InnerRadius = 1.0

[<Literal>]
let MiddleRadius = 5.0

[<Literal>]
let OuterRadius = 10.0


let score (x: double) (y: double) : int =
    match sqrt (x ** 2 + y ** 2) with
    | innerCircle when innerCircle <= InnerRadius -> 10
    | middleCircle when middleCircle <= MiddleRadius -> 5
    | outerCircle when outerCircle <= OuterRadius -> 1
    | _ -> 0

File: haskell/darts/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | radius <= innerCircle = 10
  | radius <= middleCircle = 5
  | radius <= outerCircle = 1
  | otherwise = 0
  where
    radius = sqrt (x * x + y * y)
    (innerCircle, middleCircle, outerCircle) = (1, 5, 10)
